gbpservice/neutron/nfp/configurator/api/v1/controllers/controller.py

```diff
@@ -0,0 +1,184 @@
+import json
+import subprocess
+
+from neutron.common import rpc as n_rpc
+from neutron.agent.common import config
+from oslo_config import cfg
+import oslo_messaging
+from pecan import expose, request
+from pecan import rest
+
+import constants
+
+
+class Controller(rest.RestController):
+
+    """controller class for handling all the curl request"""
+
+    def __init__(self, module_name):
+        self.host = subprocess.check_output('hostname', shell=True).rstrip()
+        self.rpcclient = RPCClient(topic=constants.TOPIC, host=self.host)
+        self.module_name = module_name
+        super(Controller, self).__init__()
+
+    @expose(method='GET', content_type='application/json')
+    def get(self):
+        try:
+            return self._get_notifications()
+        except Exception as e:
+            return json.dumps({'err_msg': e.args})
+
+    @expose(method='POST', content_type='application/json')
+    def post(self, **body):
+        try:
+            body = None
+            if request.is_body_readable:
+                body = request.json_body
+            if self.module_name == "create_network_function_device_config":
+                return self._create_network_function_device_config(body)
+            elif self.module_name == "create_network_function_config":
+                return self._create_network_function_config(body)
+            elif self.module_name == "delete_network_function_device_config":
+                return self._delete_network_function_device_config(body)
+            elif self.module_name == "delete_network_function_config":
+                return self._delete_network_function_config(body)
+            else:
+                raise Exception('Invalid Request')
+        except Exception as e:
+            return json.dumps({'err_msg': e.args})
```

Comment:
No need to return anything in case of exception for a POST request. For failures in pecan to send message to configurator, a notification should be sent back with failure reason.